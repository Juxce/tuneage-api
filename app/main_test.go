package api

import (
	"io/ioutil"
	"net/http"
	"net/http/httptest"
	"os"
	"testing"
	"time"

	approvals "github.com/approvals/go-approval-tests"
)

func TestResponse(t *testing.T) {
	req := httptest.NewRequest(http.MethodGet, "/", nil)
	w := httptest.NewRecorder()

	Data(w, req)

	approvals.VerifyJSONStruct(t, w)
}

func TestData(t *testing.T) {
	NowTimeFunc := MockNowTimeFunc
	req := httptest.NewRequest(http.MethodGet, "/", nil)
	w := httptest.NewRecorder()

	Data(w, req)

	res := w.Result()
	defer res.Body.Close()

	data, err := ioutil.ReadAll(res.Body)

	if err != nil {
		t.Errorf("Error: %v", err)
	}

	approvals.VerifyJSONBytes(t, data)
}

func MockNowTimeFunc() time.Time {
	return time.Date(2000, 12, 15, 17, 8, 00, 0, time.UTC)
}

func TestMain(m *testing.M) {
	approvals.UseFolder("testdata")
	os.Exit(m.Run())
}
