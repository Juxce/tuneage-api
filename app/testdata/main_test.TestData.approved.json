{
  "now": 976900080000
}